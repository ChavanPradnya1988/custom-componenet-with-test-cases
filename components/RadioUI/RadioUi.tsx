import React, { useState } from "react";
import { FormControl, Radio, Box, Text, VStack } from "native-base";

const RadioUi = (props: any) => {
  console.log(props);
  const [groupValue, setGroupValue] = React.useState(props.selected);
  const heading = props.data.heading;
  const description = props.data.desc;
  return (
    <Box>
      <FormControl isInvalid>
        <Radio.Group
          colorScheme="success"
          name="exampleGroup"
          accessibilityLabel="select"
          value={groupValue}
          onChange={(nextValue: any) => {
            setGroupValue(nextValue);
          }}
          m="2"
        >
          <Radio value={heading} m="2" size="sm">
            <VStack>
              <Text bold m="1">
                {heading}
              </Text>

              <Text fontSize="xs" m="1">
                {description}
              </Text>
            </VStack>
          </Radio>
        </Radio.Group>
      </FormControl>
    </Box>
  );
};

export default RadioUi;
