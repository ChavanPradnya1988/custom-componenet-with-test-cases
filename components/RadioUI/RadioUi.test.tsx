import React from "react";
import renderer from "react-test-renderer";
import { NativeBaseProvider } from "native-base";
import RadioUi from "./RadioUi";

it("expect to render radio button", () => {
  const radioButtonUiSnap = renderer
    .create(
      <NativeBaseProvider>
        <RadioUi />
      </NativeBaseProvider>
    )
    .toJSON();
  expect(radioButtonUiSnap).toMatchSnapshot();
});
