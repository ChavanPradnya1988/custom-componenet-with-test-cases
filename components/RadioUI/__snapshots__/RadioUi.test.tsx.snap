// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render radio button 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
